src/Shared/Microsoft.AspNetCore.Razor.Test.Common/Language/RazorProjectEngineTestBase.cs

```diff
@@ -1,35 +1,115 @@
 ﻿// Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
-#nullable disable
-
 using System;
 
 namespace Microsoft.AspNetCore.Razor.Language;
 
 public abstract class RazorProjectEngineTestBase
 {
+    private RazorProjectEngine? _projectEngine;
+
+    /// <summary>
+    ///  A default <see cref="RazorProjectEngine"/> instance that is configured with
+    ///  <see cref="ConfigureProjectEngine(RazorProjectEngineBuilder)"/>.
+    /// </summary>
+    protected RazorProjectEngine ProjectEngine
+    {
+        get
+        {
+            return _projectEngine ??= InterlockedOperations.Initialize(ref _projectEngine, CreateProjectEngine());
```

Comment:
Did you mean to both assign and Initialize the same field?
```suggestion
            return _projectEngine ?? InterlockedOperations.Initialize(ref _projectEngine, CreateProjectEngine());
```

Reply:
Derp! Very good catch. Thanks!